File: ubc/sqrt.ts
/**
 * https://leetcode.com/problems/sqrtx/description/
 */
export { mySqrt };

function mySqrt(x: number): number {
  if (x === 1) return 1;

  let i = 1;

  for (; i < x; i++) {
    if (i * i > x) break;
  }

  return i - 1;
}

File: returned/python/today5/sqrt.py
# https://leetcode.com/problems/sqrtx/description/

class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 0, x
        result = 0
        while l <= r:
            mid = l + ((r - l) // 2)

            if mid ** 2 > x:
                r = mid - 1
            elif mid ** 2 < x:
                l = mid + 1
                result = mid
            else:
                return mid
        return result